### ISSUE TITLE
Render method is non-differentiable

### ISSUE THREAD

**Issue by minha12** [phizaz/diffae#43]
Hi! Thanks for your amazing work. I found diff-ae is quite interesting and trying to apply it to my research as well. However, I found one of the issues that make it impossible to put an outer optimization loop with the diff-ae sampler at a simple image generator (similar to, e.g., a GAN model). In particular, I'm interested in doing manipulation the attributes with a `beta` tensor that `beta.requires_grad = True`, like this:

 `cond2 = cond * beta + cond_new * (1 - beta)`

Note that `cond` and `cond_new` are calculated beforehand without gradients. The problem I faced is that when I generate an image from the `cond2` which has gradients in this way:

`gen_img = model.render(xT, cond2, T=100)`

At this step, no gradients can flow back through `model.render()`. The optimization loop which is supposed to update `beta` cannot update weights. 

So, my question is: why the `model.render()` cannot be differential, and is there any option that allows gradients to flow back this method?

**Comment by minha12** [comment]
A deadline is closing in and I'm still struggling with this issue, so anyone who knows an idea, please help!

**Comment by phizaz** [comment]
There is `torch.no_grad` context in the code path for generation. You may want to disable these at `diffusion/base.py`. For example: https://github.com/phizaz/diffae/blob/865f1926ce0d994e4a8dc2b5b250d57f519cadc1/diffusion/base.py#L794

**Comment by minha12** [comment]
Thank you so much for your help! Your suggestion to disable the `torch.no_grad` context in `diffusion/base.py` worked perfectly and fixed the problem. 

There is one catch, though, (a note to someone who will face the same problem) with 24GB of VRAM, and the torch.no_grad() turned off, soon the VRAM will run out. A workaround is to set T = 10 or lower.

Thanks again for taking the time to help.
